### PR TITLE
br: use the correct status address

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/34651

Problem Summary:
- After we implement get config from tikv status address. we found sometimes the hostname of tikv node address and tikv status address are not same. and at this situation we should use node address' hostName and status address Port to spell the real status address.
- this pr also tried to fix the broken ci by use the correct tls http client: https://ci.pingcap.net/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/11623/pipeline/

### What is changed and how it works?
This pr tried use node address' hostName and status address Port to spell the real status address when they are not matched.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by joccau — APPROVED** [review]
LGTM

File: br/pkg/conn/conn.go [hunk 1]
```diff
@@ -545,3 +548,38 @@ func (mgr *Mgr) GetConfigFromTiKV(ctx context.Context, cli *http.Client, fn func
...
+	if err != nil {
+		return nil, err
+	}
+
+	// we try status address as default
+	addr := statusUrl
+	// but in sometimes we may not get the correct status address from PD.
+	if statusUrl.Hostname() != nodeUrl.Hostname() {
+		// if not matched, we use the address as default, but change the port
+		addr.Host = nodeUrl.Hostname() + ":" + statusUrl.Port()
```
Comment (YuJuncen) [inline br/pkg/conn/conn.go:576]: Maybe we can set `addr.Host` and `addr.Port` respectively?

**Review by YuJuncen — APPROVED** [review]
lgtm

**Pull request merged**